diff --git a/defaults/autoconfig/prefcalls.js b/defaults/autoconfig/prefcalls.js
index fcd0554..9bc7a1b 100755
--- ./defaults/autoconfig/prefcalls.js
+++ ./defaults/autoconfig/prefcalls.js
@@ -217,3 +217,5 @@ for (let [defineAs, func] of Object.entries(APIs)) {
 Object.defineProperty(Cu.waiveXrays(gSandbox), "gIsUTF8", {
   get: Cu.exportFunction(() => gIsUTF8, gSandbox),
 });
+
+Services.scriptloader.loadSubScript("resource://gre/defaults/autoconfig/prefcalls-firedragon.js");
