diff --git a/browser/components/preferences/config/passwords-autofill.mjs b/browser/components/preferences/config/passwords-autofill.mjs
index 3f614b5e20b9..42ac5e290c97 100644
--- a/browser/components/preferences/config/passwords-autofill.mjs
+++ b/browser/components/preferences/config/passwords-autofill.mjs
@@ -217,6 +217,7 @@ const FORM_AUTOFILL_CONFIG = {
         },
       },
     ],
+    hidden: Services.prefs.getBoolPref('firedragon.hidePasswdmgr'),
   },
   addresses: {
     l10nId: "addresses-group",
@@ -241,6 +242,7 @@ const FORM_AUTOFILL_CONFIG = {
         },
       },
     ],
+    hidden: Services.prefs.getBoolPref('firedragon.hidePasswdmgr'),
   },
 };
 
@@ -807,6 +809,7 @@ SettingGroupManager.registerGroups({
         ],
       },
     ],
+    hidden: Services.prefs.getBoolPref('firedragon.hidePasswdmgr'),
   },
   managePayments: {
     items: [
@@ -823,6 +826,7 @@ SettingGroupManager.registerGroups({
         },
       },
     ],
+    hidden: Services.prefs.getBoolPref('firedragon.hidePasswdmgr'),
   },
   manageAddresses: {
     items: [
@@ -839,6 +843,7 @@ SettingGroupManager.registerGroups({
         },
       },
     ],
+    hidden: Services.prefs.getBoolPref('firedragon.hidePasswdmgr'),
   },
 });
 SettingGroupManager.registerGroups(FORM_AUTOFILL_CONFIG);
