diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index b006d5e5e64c..1396026f9189 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -98,6 +98,8 @@
   <script type="module" src="chrome://browser/content/preferences/widgets/sync-device-name.mjs"></script>
   <script type="module" src="chrome://browser/content/preferences/widgets/sync-engines-list.mjs"></script>
   <script type="module" src="chrome://browser/content/ipprotection/bandwidth-usage.mjs"></script>
+
+#include ../../firedragon/content/dist/preferences.inc.xhtml
 </head>
 
 <html:body xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul"
