diff --git a/browser/components/preferences/config/privacy.mjs b/browser/components/preferences/config/privacy.mjs
index 8b67ea2..1b34787 100644
--- a/browser/components/preferences/config/privacy.mjs
+++ b/browser/components/preferences/config/privacy.mjs
@@ -2956,7 +2826,7 @@ Preferences.addSetting({
 });
 
 // Study opt out
-if (lazy.AppConstants.MOZ_DATA_REPORTING) {
+if (true) {
   Preferences.addAll([
     // Preference instances for prefs that we need to monitor while the page is open.
     { id: "app.shield.optoutstudies.enabled", type: "bool" },
