diff --git a/browser/components/preferences/preferences.js b/browser/components/preferences/preferences.js
index a76699af7525..e7745f7f3f14 100644
--- a/browser/components/preferences/preferences.js
+++ b/browser/components/preferences/preferences.js
@@ -363,6 +363,7 @@ const CONFIG_PANES = Object.freeze({
     iconSrc: "chrome://browser/skin/notification-icons/geo.svg",
     module:
       "chrome://browser/content/preferences/config/passwords-autofill.mjs",
+    visible: () => !Services.prefs.getBoolPref('firedragon.hidePasswdmgr'),
   },
   manageMemories: {
     parent: "personalizeSmartWindow",
@@ -378,6 +379,7 @@ const CONFIG_PANES = Object.freeze({
     iconSrc: "chrome://browser/skin/payment-methods-16.svg",
     module:
       "chrome://browser/content/preferences/config/passwords-autofill.mjs",
+    visible: () => !Services.prefs.getBoolPref('firedragon.hidePasswdmgr'),
   },
   profiles: {
     parent: srdSectionEnabled("sync") ? "sync" : "general",
@@ -387,7 +389,7 @@ const CONFIG_PANES = Object.freeze({
   permissionsData: {
     l10nId: "permissions-data-section",
     iconSrc: "chrome://browser/skin/permissions.svg",
-    groupIds: ["permissions", "dataCollection"],
+    groupIds: ["permissions"],
     module: "chrome://browser/content/preferences/config/permissions-data.mjs",
     visible: () => srdSectionEnabled("permissionsData"),
   },
@@ -405,7 +407,7 @@ const CONFIG_PANES = Object.freeze({
     groupIds: ["passwords", "payments", "addresses"],
     module:
       "chrome://browser/content/preferences/config/passwords-autofill.mjs",
-    visible: () => srdSectionEnabled("passwordsAutofill"),
+    visible: () => srdSectionEnabled("passwordsAutofill") && !Services.prefs.getBoolPref('firedragon.hidePasswdmgr'),
   },
   privacy: {
     l10nId: "pane-privacy-section",
