diff --git a/browser/components/sessionstore/SessionStore.sys.mjs b/browser/components/sessionstore/SessionStore.sys.mjs
index 52fc803050cd..4bda1a34d135 100644
--- a/browser/components/sessionstore/SessionStore.sys.mjs
+++ b/browser/components/sessionstore/SessionStore.sys.mjs
@@ -161,6 +161,7 @@ const lazy = XPCOMUtils.declareLazy({
   HomePage: "resource:///modules/HomePage.sys.mjs",
   JsonSchema: "resource://gre/modules/JsonSchema.sys.mjs",
   PrivacyFilter: "resource://gre/modules/sessionstore/PrivacyFilter.sys.mjs",
+  PrivateContainer: 'resource://firedragon/modules/PrivateContainer.sys.mjs',
   sessionStoreLogger:
     "moz-src:///browser/components/sessionstore/SessionLogger.sys.mjs",
   RunState: "moz-src:///browser/components/sessionstore/RunState.sys.mjs",
@@ -6964,7 +6965,7 @@ class _SessionStore {
           entryUrl == "about:privatebrowsing") &&
         !aTabState.userTypedValue
       )
-    );
+    ) && aTabState.userContextId !== lazy.PrivateContainer.userContextId;
   }
 
   /**
