diff --git a/services/settings/dumps/main/moz.build b/services/settings/dumps/main/moz.build
index 2c83ffc7874a..3e705989a6bb 100644
--- a/services/settings/dumps/main/moz.build
+++ b/services/settings/dumps/main/moz.build
@@ -26,9 +26,7 @@ if not CONFIG["MOZ_BUILD_APP"].startswith("mobile/"):
         "hijack-blocklists.json",
         "moz-essential-domain-fallbacks.json",
         "password-rules.json",
-        "search-config-icons.json",
         "search-config-overrides-v2.json",
-        "search-config-v2.json",
         "search-default-override-allowlist.json",
         "translations-models.json",
         "translations-wasm.json",
