diff --git a/toolkit/xre/nsXREDirProvider.cpp b/toolkit/xre/nsXREDirProvider.cpp
index dcce3335e14f..81b0b857219d 100644
--- a/toolkit/xre/nsXREDirProvider.cpp
+++ b/toolkit/xre/nsXREDirProvider.cpp
@@ -1337,13 +1337,11 @@ nsresult nsXREDirProvider::AppendFromAppData(nsIFile* aFile, bool aIsDotted) {
   // future?
   if (gAppData->profile) {
     nsAutoCString profile;
-#  if defined(MOZ_THUNDERBIRD)
-    if (gAppData->profile[0] != '.') {
-      profile.Assign('.');
-    }
-#  endif
-    profile.Append(gAppData->profile);
-    MOZ_TRY(aFile->AppendRelativeNativePath(profile));
+    profile = gAppData->profile;
+    ToLowerCase(profile);
+
+    MOZ_TRY(aFile->AppendRelativeNativePath(aIsDotted ? ("."_ns + profile)
+                                                      : profile));
   } else {
     nsAutoCString vendor;
     nsAutoCString appName;
@@ -1520,28 +1518,8 @@ nsresult nsXREDirProvider::GetLegacyOrXDGHomePath(const char* aHomeDir,
       return NS_OK;
     }
 
-    // If the build was made against a specific profile name, MOZ_APP_PROFILE=
-    // then make sure we respect this and dont move to XDG directory
-    //
-    // Only do this on Firefox.
-    //
-    // For Thunderbird, it always defines MOZ_APP_PROFILE=thunderbird. Hence,
-    // check if the gAppData->profile value is set to something else, indicating
-    // of a specific build where legacy needs to be forced, similar to the
-    // Firefox handling. Both casing are verified to account for systems where
-    // that matters.
-    if (gAppData->profile
-#  if defined(MOZ_THUNDERBIRD)
-        && strcmp(gAppData->profile, "thunderbird") != 0 &&
-        strcmp(gAppData->profile, "Thunderbird") != 0
-#  endif
-    ) {
-      MOZ_TRY(NS_NewNativeLocalFile(nsDependentCString(aHomeDir),
-                                    getter_AddRefs(localDir)));
-    } else {
-      MOZ_TRY(GetLegacyOrXDGConfigHome(aHomeDir, getter_AddRefs(localDir)));
-      MOZ_TRY(localDir->Clone(getter_AddRefs(parentDir)));
-    }
+    MOZ_TRY(GetLegacyOrXDGConfigHome(aHomeDir, getter_AddRefs(localDir)));
+    MOZ_TRY(localDir->Clone(getter_AddRefs(parentDir)));
 
     MOZ_TRY(AppendFromAppData(localDir, false));
   }
