diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index 2b4d35e060..7fcd7f6048 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -166,6 +166,23 @@
 @BINPATH@/@DLL_PREFIX@mozavutil@DLL_SUFFIX@
 @BINPATH@/@DLL_PREFIX@mozavcodec@DLL_SUFFIX@
 
+; [Noraneko Files]
+#ifdef NORANEKO_DIST
+@RESPATH@/chrome/noraneko.manifest
+@RESPATH@/chrome/noraneko@JAREXT@
+#endif
+#ifdef NORANEKO_BUILDID2
+@RESPATH@/browser/buildid2
+#endif
+
+; [FireDragon Files]
+@RESPATH@/chrome/firedragon.manifest
+@RESPATH@/chrome/firedragon@JAREXT@
+#ifdef FIREDRAGON_SETTINGS
+@RESPATH@/distribution/policies.json
+@RESPATH@/firedragon.cfg
+#endif
+
 ; [Components]
 #ifdef ACCESSIBILITY
 #ifdef XP_WIN
@@ -384,10 +401,6 @@ bin/libfreebl_64int_3.so
 @BINPATH@/crashhelper@BIN_SUFFIX@
 #endif
 
-; [ Ping Sender ]
-;
-@BINPATH@/pingsender@BIN_SUFFIX@
-
 ; [ Native Messaging Host Proxy ]
 ;
 #if defined(XP_WIN) || defined(XP_MACOSX)
